lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -147,11 +147,23 @@ bool TempRValueOptPass::collectLoads(
     LLVM_DEBUG(llvm::dbgs()
                << "  Temp use may write/destroy its source" << *user);
     return false;
-
   case SILInstructionKind::BeginAccessInst:
     return cast<BeginAccessInst>(user)->getAccessKind() == SILAccessKind::Read;
 
+  case SILInstructionKind::MarkDependenceInst: {
```

Comment:
this looks correct, but please comment why mark_dependence base operand is safe to consider a load, and why the value operand needs to be transitive (otherwise we would miscompile!)

Reply:
Sorry I missed this. Will update as soon as my ci jobs turn green. Don't want to reset state yet.